CrossPlatformDesktopProject/CrossPlatformDesktopProject/CollisionHandler/CollisionResponse.cs

```diff
@@ -0,0 +1,117 @@
+﻿using CrossPlatformDesktopProject.Commands;
+using CrossPlatformDesktopProject.Equipables;
+using CrossPlatformDesktopProject.Link;
+using CrossPlatformDesktopProject.NPC;
+using Microsoft.Xna.Framework;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Security.Cryptography.X509Certificates;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace CrossPlatformDesktopProject.CollisionHandler
+{
+    class CollisionResponse
+    {
+        private Dictionary<Tuple<Type, Type, CollisionSides>, Type> commandMap;
+        private HashSet<Tuple<Type, Type, CollisionSides>> keySet;
+
+        public CollisionResponse()
+        {
+            commandMap = new Dictionary<Tuple<Type, Type, CollisionSides>, Type>();
+            BuildDictionary();
+            keySet = new HashSet<Tuple<Type, Type, CollisionSides>>(commandMap.Keys);
+        }
+
+        private void BuildDictionary()
+        {
+            Type playerType = typeof(Player);
+            
+            //Enemy Types
+            Type batType = typeof(Bat);
+            Type bossType = typeof(Boss);
+            Type gelType = typeof(Gel);
+            Type goriyaType = typeof(Goriya);
+            Type skeletonType = typeof(Skeleton);
+            Type[] enemyTypes = { batType, bossType, gelType, goriyaType, skeletonType };
```

Comment:
Wouldn't something like this work and be less code?

```suggestion
            Type[] enemyTypes = {typeof(Bat), typeof(Boss), typeof(Gel), typeof(Goriya), typeof(Skeleton)};
```

Reply:
Yes, but different enemies may have different interactions, so I'm still hesitant to reduce any of these.